### ISSUE TITLE
When building Pose for Python GPU, " Unable to find the type for stream "segmentation_mask"." would happen. 

### ISSUE THREAD

**Issue by FalschMico** [google/mediapipe#3125]
I'm trying to build gpu version for python on ubuntu 20.04, but can't run the pose sample with gpu. The error is below:
`RuntimeError: ; Unable to find the type for stream "segmentation_mask".  It may be set to AnyType or something else that isn't determinable, or the type may be defined but not registered.`
 **Before that, I've built GPU version of hands with success, so I built pose in the similar way.**
- In https://github.com/google/mediapipe/blob/master/mediapipe/modules/pose_landmark/pose_landmark_gpu.pbtxt#L132, I added
```Python
node {
  calculator: "ColorConvertCalculator"
  input_stream: "RGB_IN:image"
  output_stream: "RGBA_OUT:image_rgba"
}

node {
  calculator: "ImageFrameToGpuBufferCalculator"
  input_stream: "image_rgba"
  output_stream: "image_gpu"
}
```
I guess somthing wrong exists in some calculator related to **segmentation_mask**.  Anyone can help?

**Comment by FalschMico** [comment]
> Hi @FalschMico , could you provide complete steps to reproduce the issue.

@sureshdagooglecom 
- what I 've done in hands build:
	- I followed those two documents sequently.
		[https://google.github.io/mediapipe/getting_started/install.html](https://google.github.io/mediapipe/getting_started/install.html)
		 [https://google.github.io/mediapipe/getting_started/python.html](https://google.github.io/mediapipe/getting_started/python.html)
		Everything was OK, and I made the cpu version successfully to run the sample.
	- Modify [`setup.py`](https://github.com/google/mediapipe/blob/master/setup.py)
		- change line 217-227 to
			```Python
			 def run(self):
			    _check_bazel()
			    binary_graphs = [
			        'face_detection/face_detection_short_range_gpu',
			        'face_detection/face_detection_full_range_gpu',
			        'face_landmark/face_landmark_front_gpu',
			        'hand_landmark/hand_landmark_tracking_gpu',
			        'holistic_landmark/holistic_landmark_gpu', 'objectron/objectron_gpu',
			        'pose_landmark/pose_landmark_gpu',
			        'selfie_segmentation/selfie_segmentation_gpu',
			        'palm_detection/palm_detection_gpu',
			    ]
			```
			
		- change line 236-244
			```Python
			 bazel_command = [
			        'bazel',
			        'build',
			        '--compilation_mode=opt',
			        '--copt=-DNDEBUG',
			        '--copt=-DMESA_EGL_NO_X11_HEADERS',
			        '--copt=-DEGL_NO_X11',
			        '--action_env=PYTHON_BIN_PATH=' + _normalize_path(sys.executable),
			        os.path.join('mediapipe/modules/', graph_path),
			    ]
			```
			
		- change line 293-300
			```Python
			    bazel_command = [
			        'bazel',
			        'build',
			        '--compilation_mode=opt',
			        '--copt=-DNDEBUG',
			        '--copt=-DMESA_EGL_NO_X11_HEADERS',
			        '--copt=-DEGL_NO_X11',
			        '--action_env=PYTHON_BIN_PATH=' + _normalize_path(sys.executable),
			        str(ext.bazel_target + '.so'),
			    ]
			```
			
		- Modify [`python/BUILD`](https://github.com/google/mediapipe/blob/master/mediapipe/python/BUILD)like below:
			```Python
			cc_library(
			    name = "builtin_calculators",
			    deps = [
			        "//mediapipe/calculators/core:gate_calculator",
			        "//mediapipe/calculators/core:pass_through_calculator",
			        "//mediapipe/calculators/core:side_packet_to_stream_calculator",
			        "//mediapipe/calculators/core:split_landmarks_calculator",
			        "//mediapipe/calculators/core:string_to_int_calculator",
			        "//mediapipe/calculators/image:image_transformation_calculator",
			        "//mediapipe/calculators/util:detection_unique_id_calculator",
			        "//mediapipe/modules/face_detection:face_detection_full_range_gpu",
			        "//mediapipe/modules/face_detection:face_detection_short_range_gpu",
			        "//mediapipe/modules/face_landmark:face_landmark_front_gpu",
			        "//mediapipe/modules/hand_landmark:hand_landmark_tracking_gpu",
			        "//mediapipe/modules/holistic_landmark:holistic_landmark_gpu",
			        "//mediapipe/modules/objectron:objectron_gpu",
			        "//mediapipe/modules/palm_detection:palm_detection_gpu",
			        "//mediapipe/modules/pose_detection:pose_detection_gpu",
			        "//mediapipe/modules/pose_landmark:pose_landmark_by_roi_gpu",
			        "//mediapipe/modules/pose_landmark:pose_landmark_gpu",
			        "//mediapipe/modules/selfie_segmentation:selfie_segmentation_gpu",
			        "//mediapipe/gpu:gpu_buffer_to_image_frame_calculator",
			        "//mediapipe/gpu:image_frame_to_gpu_buffer_calculator",
			        "//mediapipe/calculators/image:color_convert_calculator"
			    ],
			```
			
		- Modify[` python/solutions/pose.py`](https://github.com/google/mediapipe/blob/master/mediapipe/python/solutions/pose.py#L90) line 90:
			`_BINARYPB_FILE_PATH = 'mediapipe/modules/pose_landmark/pose_landmark_gpu.binarypb'`
		- Modify[` hand_landmark_tracking_gpu.pbtxt`](https://github.com/google/mediapipe/blob/master/mediapipe/modules/hand_landmark/hand_landmark_tracking_gpu.pbtxt) according to #3105
		<details>
                <summary>hand_landmark/hand_landmark_tracking_gpu.pbtxt content</summary>
			```Python

			
			type: "HandLandmarkTrackingGpu"
			
			# GPU image. (GpuBuffer)
			input_stream: "IMAGE:image"
			
			# Max number of hands to detect/track. (int)
			input_side_packet: "NUM_HANDS:num_hands"
			
			# Complexity of hand landmark and palm detection models: 0 or 1. Accuracy as
			# well as inference latency generally go up with the model complexity. If
			# unspecified, functions as set to 1. (int)
			input_side_packet: "MODEL_COMPLEXITY:model_complexity"
			
			# Whether landmarks on the previous image should be used to help localize
			# landmarks on the current image. (bool)
			input_side_packet: "USE_PREV_LANDMARKS:use_prev_landmarks"
			
			# Collection of detected/predicted hands, each represented as a list of
			# landmarks. (std::vector<NormalizedLandmarkList>)
			# NOTE: there will not be an output packet in the LANDMARKS stream for this
			# particular timestamp if none of hands detected. However, the MediaPipe
			# framework will internally inform the downstream calculators of the absence of
			# this packet so that they don't wait for it unnecessarily.
			output_stream: "LANDMARKS:multi_hand_landmarks"
			
			# Collection of detected/predicted hand world landmarks.
			# (std::vector<LandmarkList>)
			#
			# World landmarks are real-world 3D coordinates in meters with the origin in the
			# center of the hand bounding box calculated from the landmarks.
			#
			# WORLD_LANDMARKS shares the same landmark topology as LANDMARKS. However,
			# LANDMARKS provides coordinates (in pixels) of a 3D object projected onto the
			# 2D image surface, while WORLD_LANDMARKS provides coordinates (in meters) of
			# the 3D object itself.
			output_stream: "WORLD_LANDMARKS:multi_hand_world_landmarks"
			
			# Collection of handedness of the detected hands (i.e. is hand left or right),
			# each represented as a ClassificationList proto with a single Classification
			# entry. (std::vector<ClassificationList>)
			# Note that handedness is determined assuming the input image is mirrored,
			# i.e., taken with a front-facing/selfie camera with images flipped
			# horizontally.
			output_stream: "HANDEDNESS:multi_handedness"
			
			# Extra outputs (for debugging, for instance).
			# Detected palms. (std::vector<Detection>)
			output_stream: "PALM_DETECTIONS:palm_detections"
			# Regions of interest calculated based on landmarks.
			# (std::vector<NormalizedRect>)
			output_stream: "HAND_ROIS_FROM_LANDMARKS:hand_rects"
			# Regions of interest calculated based on palm detections.
			# (std::vector<NormalizedRect>)
			output_stream: "HAND_ROIS_FROM_PALM_DETECTIONS:hand_rects_from_palm_detections"
			
			# When the optional input side packet "use_prev_landmarks" is either absent or
			# set to true, uses the landmarks on the previous image to help localize
			# landmarks on the current image.
			node {
			  calculator: "GateCalculator"
			  input_side_packet: "ALLOW:use_prev_landmarks"
			  input_stream: "prev_hand_rects_from_landmarks"
			  output_stream: "gated_prev_hand_rects_from_landmarks"
			  options: {
			    [mediapipe.GateCalculatorOptions.ext] {
			      allow: true
			    }
			  }
			}
			
			# Determines if an input vector of NormalizedRect has a size greater than or
			# equal to the provided num_hands.
			node {
			  calculator: "NormalizedRectVectorHasMinSizeCalculator"
			  input_stream: "ITERABLE:gated_prev_hand_rects_from_landmarks"
			  input_side_packet: "num_hands"
			  output_stream: "prev_has_enough_hands"
			}
			
			node {
			  calculator: "ColorConvertCalculator"
			  input_stream: "RGB_IN:image"
			  output_stream: "RGBA_OUT:image_rgba"
			}
			
			node {
			  calculator: "ImageFrameToGpuBufferCalculator"
			  input_stream: "image_rgba"
			  output_stream: "image_gpu"
			}
			
			
			# Drops the incoming image if enough hands have already been identified from the
			# previous image. Otherwise, passes the incoming image through to trigger a new
			# round of palm detection.
			node {
			  calculator: "GateCalculator"
			  input_stream: "image_gpu"
			  input_stream: "DISALLOW:prev_has_enough_hands"
			  output_stream: "palm_detection_image"
			  options: {
			    [mediapipe.GateCalculatorOptions.ext] {
			      empty_packets_as_allow: true
			    }
			  }
			}
			
			# Detects palms.
			node {
			  calculator: "PalmDetectionGpu"
			  input_side_packet: "MODEL_COMPLEXITY:model_complexity"
			  input_stream: "IMAGE:palm_detection_image"
			  output_stream: "DETECTIONS:all_palm_detections"
			}
			
			# Makes sure there are no more detections than provided num_hands.
			node {
			  calculator: "ClipDetectionVectorSizeCalculator"
			  input_stream: "all_palm_detections"
			  output_stream: "palm_detections"
			  input_side_packet: "num_hands"
			}
			
			# Extracts image size.
			node {
			  calculator: "ImagePropertiesCalculator"
			  input_stream: "IMAGE_GPU:palm_detection_image"
			  output_stream: "SIZE:palm_detection_image_size"
			}
			
			# Outputs each element of palm_detections at a fake timestamp for the rest of
			# the graph to process. Clones the image_size packet for each palm_detection at
			# the fake timestamp. At the end of the loop, outputs the BATCH_END timestamp
			# for downstream calculators to inform them that all elements in the vector have
			# been processed.
			node {
			  calculator: "BeginLoopDetectionCalculator"
			  input_stream: "ITERABLE:palm_detections"
			  input_stream: "CLONE:palm_detection_image_size"
			  output_stream: "ITEM:palm_detection"
			  output_stream: "CLONE:image_size_for_palms"
			  output_stream: "BATCH_END:palm_detections_timestamp"
			}
			
			# Calculates region of interest (ROI) base on the specified palm.
			node {
			  calculator: "PalmDetectionDetectionToRoi"
			  input_stream: "DETECTION:palm_detection"
			  input_stream: "IMAGE_SIZE:image_size_for_palms"
			  output_stream: "ROI:hand_rect_from_palm_detection"
			}
			
			# Collects a NormalizedRect for each hand into a vector. Upon receiving the
			# BATCH_END timestamp, outputs the vector of NormalizedRect at the BATCH_END
			# timestamp.
			node {
			  name: "EndLoopForPalmDetections"
			  calculator: "EndLoopNormalizedRectCalculator"
			  input_stream: "ITEM:hand_rect_from_palm_detection"
			  input_stream: "BATCH_END:palm_detections_timestamp"
			  output_stream: "ITERABLE:hand_rects_from_palm_detections"
			}
			
			# Performs association between NormalizedRect vector elements from previous
			# image and rects based on palm detections from the current image. This
			# calculator ensures that the output hand_rects vector doesn't contain
			# overlapping regions based on the specified min_similarity_threshold.
			node {
			  calculator: "AssociationNormRectCalculator"
			  input_stream: "hand_rects_from_palm_detections"
			  input_stream: "gated_prev_hand_rects_from_landmarks"
			  output_stream: "hand_rects"
			  options: {
			    [mediapipe.AssociationCalculatorOptions.ext] {
			      min_similarity_threshold: 0.5
			    }
			  }
			}
			
			# Extracts image size.
			node {
			  calculator: "ImagePropertiesCalculator"
			  input_stream: "IMAGE_GPU:image_gpu"
			  output_stream: "SIZE:image_size"
			}
			
			# Outputs each element of hand_rects at a fake timestamp for the rest of the
			# graph to process. Clones image and image size packets for each
			# single_hand_rect at the fake timestamp. At the end of the loop, outputs the
			# BATCH_END timestamp for downstream calculators to inform them that all
			# elements in the vector have been processed.
			node {
			  calculator: "BeginLoopNormalizedRectCalculator"
			  input_stream: "ITERABLE:hand_rects"
			  input_stream: "CLONE:0:image_gpu"
			  input_stream: "CLONE:1:image_size"
			  output_stream: "ITEM:single_hand_rect"
			  output_stream: "CLONE:0:image_for_landmarks"
			  output_stream: "CLONE:1:image_size_for_landmarks"
			  output_stream: "BATCH_END:hand_rects_timestamp"
			}
			
			# Detect hand landmarks for the specific hand rect.
			node {
			  calculator: "HandLandmarkGpu"
			  input_side_packet: "MODEL_COMPLEXITY:model_complexity"
			  input_stream: "IMAGE:image_for_landmarks"
			  input_stream: "ROI:single_hand_rect"
			  output_stream: "LANDMARKS:single_hand_landmarks"
			  output_stream: "WORLD_LANDMARKS:single_hand_world_landmarks"
			  output_stream: "HANDEDNESS:single_handedness"
			}
			
			# Collects the handedness for each single hand into a vector. Upon receiving the
			# BATCH_END timestamp, outputs a vector of ClassificationList at the BATCH_END
			# timestamp.
			node {
			  calculator: "EndLoopClassificationListCalculator"
			  input_stream: "ITEM:single_handedness"
			  input_stream: "BATCH_END:hand_rects_timestamp"
			  output_stream: "ITERABLE:multi_handedness"
			}
			
			# Calculate region of interest (ROI) based on detected hand landmarks to reuse
			# on the subsequent runs of the graph.
			node {
			  calculator: "HandLandmarkLandmarksToRoi"
			  input_stream: "IMAGE_SIZE:image_size_for_landmarks"
			  input_stream: "LANDMARKS:single_hand_landmarks"
			  output_stream: "ROI:single_hand_rect_from_landmarks"
			}
			
			# Collects a set of landmarks for each hand into a vector. Upon receiving the
			# BATCH_END timestamp, outputs the vector of landmarks at the BATCH_END
			# timestamp.
			node {
			  calculator: "EndLoopNormalizedLandmarkListVectorCalculator"
			  input_stream: "ITEM:single_hand_landmarks"
			  input_stream: "BATCH_END:hand_rects_timestamp"
			  output_stream: "ITERABLE:multi_hand_landmarks"
			}
			
			# Collects a set of world landmarks for each hand into a vector. Upon receiving
			# the BATCH_END timestamp, outputs the vector of landmarks at the BATCH_END
			# timestamp.
			node {
			  calculator: "EndLoopLandmarkListVectorCalculator"
			  input_stream: "ITEM:single_hand_world_landmarks"
			  input_stream: "BATCH_END:hand_rects_timestamp"
			  output_stream: "ITERABLE:multi_hand_world_landmarks"
			}
			
			# Collects a NormalizedRect for each hand into a vector. Upon receiving the
			# BATCH_END timestamp, outputs the vector of NormalizedRect at the BATCH_END
			# timestamp.
			node {
			  calculator: "EndLoopNormalizedRectCalculator"
			  input_stream: "ITEM:single_hand_rect_from_landmarks"
			  input_stream: "BATCH_END:hand_rects_timestamp"
			  output_stream: "ITERABLE:hand_rects_from_landmarks"
			}
			
			# Caches hand rects calculated from landmarks, and upon the arrival of the next
			# input image, sends out the cached rects with timestamps replaced by that of
			# the input image, essentially generating a packet that carries the previous
			# hand rects. Note that upon the arrival of the very first input image, a
			# timestamp bound update occurs to jump start the feedback loop.
			node {
			  calculator: "PreviousLoopbackCalculator"
			  input_stream: "MAIN:image_gpu"
			  input_stream: "LOOP:hand_rects_from_landmarks"
			  input_stream_info: {
			    tag_index: "LOOP"
			    back_edge: true
			  }
			  output_stream: "PREV_LOOP:prev_hand_rects_from_landmarks"
			}
			```
</details>
			
<details>	
<summary>Then, based on these above, I tried to build **pose**. I modified :</summary>[https://github.com/google/mediapipe/blob/master/mediapipe/modules/pose_landmark/pose_landmark_gpu.pbtxt](https://github.com/google/mediapipe/blob/master/mediapipe/modules/pose_landmark/pose_landmark_gpu.pbtxt)
	
     ```Python
	
	type: "PoseLandmarkGpu"
	
	# GPU image. (GpuBuffer)
	input_stream: "IMAGE:image"
	
	# Whether to filter landmarks across different input images to reduce jitter.
	# If unspecified, functions as set to true. (bool)
	input_side_packet: "SMOOTH_LANDMARKS:smooth_landmarks"
	
	# Whether to predict the segmentation mask. If unspecified, functions as set to
	# false. (bool)
	input_side_packet: "ENABLE_SEGMENTATION:enable_segmentation"
	
	# Whether to filter segmentation mask across different input images to reduce
	# jitter. If unspecified, functions as set to true. (bool)
	input_side_packet: "SMOOTH_SEGMENTATION:smooth_segmentation"
	
	# Complexity of the pose landmark model: 0, 1 or 2. Landmark accuracy as well as
	# inference latency generally go up with the model complexity. If unspecified,
	# functions as set to 1. (int)
	input_side_packet: "MODEL_COMPLEXITY:model_complexity"
	
	# Whether landmarks on the previous image should be used to help localize
	# landmarks on the current image. (bool)
	input_side_packet: "USE_PREV_LANDMARKS:use_prev_landmarks"
	
	# Pose landmarks. (NormalizedLandmarkList)
	# We have 33 landmarks (see pose_landmark_topology.svg), and there are other


	# NOTE: if a pose is not present within the given ROI, for this particular
	# timestamp there will not be an output packet in the LANDMARKS stream. However,
	# the MediaPipe framework will internally inform the downstream calculators of
	# the absence of this packet so that they don't wait for it unnecessarily.
	output_stream: "LANDMARKS:pose_landmarks"
	
	# Pose world landmarks. (LandmarkList)
	# World landmarks are real-world 3D coordinates in meters with the origin at the
	# center between hips. WORLD_LANDMARKS shares the same landmark topology as
	# LANDMARKS. However, LANDMARKS provides coordinates (in pixels) of a 3D object
	# projected onto the 2D image surface, while WORLD_LANDMARKS provides
	# coordinates (in meters) of the 3D object itself.
	output_stream: "WORLD_LANDMARKS:pose_world_landmarks"
	
	# Segmentation mask. (GpuBuffer in RGBA, with the same mask values in R and A)
	output_stream: "SEGMENTATION_MASK:segmentation_mask"
	
	# Extra outputs (for debugging, for instance).
	# Detected poses. (Detection)
	output_stream: "DETECTION:pose_detection"
	# Regions of interest calculated based on landmarks. (NormalizedRect)
	output_stream: "ROI_FROM_LANDMARKS:pose_rect_from_landmarks"
	# Regions of interest calculated based on pose detections. (NormalizedRect)
	output_stream: "ROI_FROM_DETECTION:pose_rect_from_detection"
	
	# When the optional input side packet "use_prev_landmarks" is either absent or
	# set to true, uses the landmarks on the previous image to help localize
	# landmarks on the current image.
	node {
	  calculator: "GateCalculator"
	  input_side_packet: "ALLOW:use_prev_landmarks"
	  input_stream: "prev_pose_rect_from_landmarks"
	  output_stream: "gated_prev_pose_rect_from_landmarks"
	  options: {
	    [mediapipe.GateCalculatorOptions.ext] {
	      allow: true
	    }
	  }
	}
	
	# Checks if there's previous pose rect calculated from landmarks.
	node: {
	  calculator: "PacketPresenceCalculator"
	  input_stream: "PACKET:gated_prev_pose_rect_from_landmarks"
	  output_stream: "PRESENCE:prev_pose_rect_from_landmarks_is_present"
	}
	node {
	  calculator: "ColorConvertCalculator"
	  input_stream: "RGB_IN:image"
	  output_stream: "RGBA_OUT:image_rgba"
	}
	
	node {
	  calculator: "ImageFrameToGpuBufferCalculator"
	  input_stream: "image_rgba"
	  output_stream: "image_gpu"
	}
	# Calculates size of the image.
	node {
	  calculator: "ImagePropertiesCalculator"
	  input_stream: "IMAGE_GPU:image_gpu"
	  output_stream: "SIZE:image_size"
	}
	
	# Drops the incoming image if the pose has already been identified from the
	# previous image. Otherwise, passes the incoming image through to trigger a new
	# round of pose detection.
	node {
	  calculator: "GateCalculator"
	  input_stream: "image_gpu"
	  input_stream: "image_size"
	  input_stream: "DISALLOW:prev_pose_rect_from_landmarks_is_present"
	  output_stream: "image_for_pose_detection"
	  output_stream: "image_size_for_pose_detection"
	  options: {
	    [mediapipe.GateCalculatorOptions.ext] {
	      empty_packets_as_allow: true
	    }
	  }
	}
	
	# Detects poses.
	node {
	  calculator: "PoseDetectionGpu"
	  input_stream: "IMAGE:image_for_pose_detection"
	  output_stream: "DETECTIONS:pose_detections"
	}
	
	# Gets the very first detection from "pose_detections" vector.
	node {
	  calculator: "SplitDetectionVectorCalculator"
	  input_stream: "pose_detections"
	  output_stream: "pose_detection"
	  options: {
	    [mediapipe.SplitVectorCalculatorOptions.ext] {
	      ranges: { begin: 0 end: 1 }
	      element_only: true
	    }
	  }
	}
	
	# Calculates region of interest based on pose detection, so that can be used
	# to detect landmarks.
	node {
	  calculator: "PoseDetectionToRoi"
	  input_stream: "DETECTION:pose_detection"
	  input_stream: "IMAGE_SIZE:image_size_for_pose_detection"
	  output_stream: "ROI:pose_rect_from_detection"
	}
	
	# Selects either pose rect (or ROI) calculated from detection or from previously
	# detected landmarks if available (in this case, calculation of pose rect from
	# detection is skipped).
	node {
	  calculator: "MergeCalculator"
	  input_stream: "pose_rect_from_detection"
	  input_stream: "gated_prev_pose_rect_from_landmarks"
	  output_stream: "pose_rect"
	}
	
	# Detects pose landmarks within specified region of interest of the image.
	node {
	  calculator: "PoseLandmarkByRoiGpu"
	  input_side_packet: "MODEL_COMPLEXITY:model_complexity"
	  input_side_packet: "ENABLE_SEGMENTATION:enable_segmentation"
	  input_stream: "IMAGE:image_gpu"
	  input_stream: "ROI:pose_rect"
	  output_stream: "LANDMARKS:unfiltered_pose_landmarks"
	  output_stream: "AUXILIARY_LANDMARKS:unfiltered_auxiliary_landmarks"
	  output_stream: "WORLD_LANDMARKS:unfiltered_world_landmarks"
	  output_stream: "SEGMENTATION_MASK:unfiltered_segmentation_mask"
	}
	
	# Smoothes landmarks to reduce jitter.
	node {
	  calculator: "PoseLandmarkFiltering"
	  input_side_packet: "ENABLE:smooth_landmarks"
	  input_stream: "IMAGE_SIZE:image_size"
	  input_stream: "NORM_LANDMARKS:unfiltered_pose_landmarks"
	  input_stream: "AUX_NORM_LANDMARKS:unfiltered_auxiliary_landmarks"
	  input_stream: "WORLD_LANDMARKS:unfiltered_world_landmarks"
	  output_stream: "FILTERED_NORM_LANDMARKS:pose_landmarks"
	  output_stream: "FILTERED_AUX_NORM_LANDMARKS:auxiliary_landmarks"
	  output_stream: "FILTERED_WORLD_LANDMARKS:pose_world_landmarks"
	}
	
	# Calculates region of interest based on the auxiliary landmarks, to be used in
	# the subsequent image.
	node {
	  calculator: "PoseLandmarksToRoi"
	  input_stream: "LANDMARKS:auxiliary_landmarks"
	  input_stream: "IMAGE_SIZE:image_size"
	  output_stream: "ROI:pose_rect_from_landmarks"
	}
	
	# Caches pose rects calculated from landmarks, and upon the arrival of the next
	# input image, sends out the cached rects with timestamps replaced by that of
	# the input image, essentially generating a packet that carries the previous
	# pose rects. Note that upon the arrival of the very first input image, a
	# timestamp bound update occurs to jump start the feedback loop.
	node {
	  calculator: "PreviousLoopbackCalculator"
	  input_stream: "MAIN:image_gpu"
	  input_stream: "LOOP:pose_rect_from_landmarks"
	  input_stream_info: {
	    tag_index: "LOOP"
	    back_edge: true
	  }
	  output_stream: "PREV_LOOP:prev_pose_rect_from_landmarks"
	}
	
	# Smoothes segmentation to reduce jitter.
	node {
	  calculator: "PoseSegmentationFiltering"
	  input_side_packet: "ENABLE:smooth_segmentation"
	  input_stream: "SEGMENTATION_MASK:unfiltered_segmentation_mask"
	  output_stream: "FILTERED_SEGMENTATION_MASK:filtered_segmentation_mask"
	}
	
	# Converts the incoming segmentation mask represented as an Image into the
	# corresponding GpuBuffer type.
	node: {
	  calculator: "FromImageCalculator"
	  input_stream: "IMAGE:filtered_segmentation_mask"
	  output_stream: "IMAGE_GPU:segmentation_mask"
	}
	
	```
</details>

- And the error is:
	```Python
	RuntimeError: ; Unable to find the type for stream "segmentation_mask". It may be set to AnyType or something else that isn't determinable, or the type may be defined but not registered.
	```

**Comment by sureshdagooglecom** [comment]
Hi @FalschMico ,Have you tried checking this issue https://github.com/google/mediapipe/issues/1936#issuecomment-830382471 information w.r.t GPU

**Comment by tonywangziteng** [comment]
Is it working now? I am in face of the same problem.

**Comment by Theo-Cheynel** [comment]
I am facing the same issue, built the GPU version with those same steps, and I get the exact same error. Has any one found the solution yet ?

**Comment by hujb48** [comment]
Hi, sir. Have you found any solution about this problem yet? @FalschMico

**Comment by FalschMico** [comment]
@hujb48  Nope. I have almost given it up.

**Comment by hujb48** [comment]
> @hujb48 Nope. I have almost given it up.

@FalschMico Sorry to hear that, and I would continue to maintain attention about this.
Please keep in touch where exists any update of the issue if possible.

**Comment by FalschMico** [comment]
@hujb48 Yeah. I wish for the further progress.

**Comment by sureshdagooglecom** [comment]
MediaPipe Python package only contains CPU graphs. If you need to run any GPU graphs, see https://github.com/google/mediapipe/issues/1651 (comment).

**Comment by naveenkulkarni-movidius** [comment]
Hello, I am getting the same error as well when I try to run the v0.8.9 version of the holistic model. Since Holistic internally calls pose landmark model, likely there is registration for segmentation_mask missing. Can you help point out what is missing w.r.t segmentation_mask registration and where it needs to be registered? Does it need to be registered in mediapipe/python/BUILD? : 
        "//mediapipe/framework:basic_types_registration",
        "//mediapipe/framework/formats:classification_registration",
        "//mediapipe/framework/formats:detection_registration",
        "//mediapipe/framework/formats:landmark_registration",
        "//mediapipe/framework/formats:rect_registration",
        #"//mediapipe/modules/holistic_landmark/calculators:roi_registration",
        "//mediapipe/modules/objectron/calculators:annotation_registration", 

This is the error I see, 
File "~/anaconda3/envs/v0.8.9/lib/python3.8/site-packages/mediapipe/python/solution_base.py", line 404, in get_stream_packet_type
    validated_graph.registered_stream_type_name(stream_name))
RuntimeError: ; Unable to find the type for stream "segmentation_mask".

**Comment by Xu-ht** [comment]
I am facing the same problem when building v0.8.10.1. Have anyone found solutions yet?
`RuntimeError: ; Unable to find the type for stream "segmentation_mask".  It may be set to AnyType or something else that isn't determinable, or the type may be defined but not registered.`

**Comment by Xu-ht** [comment]
I am facing the same problem when building v0.8.10.1. Have anyone found solutions yet?
`RuntimeError: ; Unable to find the type for stream "segmentation_mask".  It may be set to AnyType or something else that isn't determinable, or the type may be defined but not registered.`

**Comment by Xu-ht** [comment]
I solved this problem on my Jeston Nano, Ubuntu 18.04. The holistic process now works with 15 fps (3 fps with CPU only in my practice). 
Modify [https://github.com/google/mediapipe/blob/master/mediapipe/modules/pose_landmark/pose_landmark_gpu.pbtxt](url):

change
```
# Converts the incoming segmentation mask represented as an Image into the
# corresponding GpuBuffer type.
node: {
  calculator: "FromImageCalculator"
  input_stream: "IMAGE:filtered_segmentation_mask"
  output_stream: "IMAGE_GPU:segmentation_mask"
}
```
to
```
# Converts the incoming segmentation mask represented as an Image into the
# corresponding GpuBuffer type.
node: {
  calculator: "FromImageCalculator"
  input_stream: "IMAGE:filtered_segmentation_mask"
  output_stream: "IMAGE_CPU:segmentation_mask"
}
```
